scripts/generate-artifacts.ps1

```
@@ -1,75 +1,118 @@
 
-Write-Output "Generating a SAS token for the template artifact"
-$date = Get-Date
-$date = $date.AddDays(30) 
-$resourcesTemplateSaSToken = New-AzStorageBlobSASToken -Container $containerName -Blob "$taskName/$resourcesTemplateName" -Permission r -ExpiryTime $date -Context $storageAccount.Context
-$resourcesTemplateURL = "$($blob.ICloudBlob.uri.AbsoluteUri)?$resourcesTemplateSaSToken"
+# 7) SAS generation with fallback
+$expiry = (Get-Date).ToUniversalTime().AddDays(30)
+$sasUrl = $null
+Write-Host "Generating SAS token..."
+try {
+  $sasUrl = New-AzStorageBlobSASToken -Context $ctx -Container $containerName -Blob $blobPath -Permission r -ExpiryTime $expiry -FullUri
+} catch {
+  Write-Warning "Implicit context SAS failed. Trying with account key..."
+  try {
+    $acctKey = (Get-AzStorageAccountKey -ResourceGroupName $resourceGroup -Name $ArtifactsStorageAccountName | Select-Object -First 1).Value
+    if (-not $acctKey) { throw "No storage account key returned." }
+    $ctxWithKey = New-AzStorageContext -StorageAccountName $ArtifactsStorageAccountName -StorageAccountKey $acctKey
+    $sasUrl = New-AzStorageBlobSASToken -Context $ctxWithKey -Container $containerName -Blob $blobPath -Permission r -ExpiryTime $expiry -FullUri
+  } catch {
+    throw "Unable to generate SAS URL. Ensure permissions to read storage keys. Details: $($_.Exception.Message)"
+  }
+}
+Write-Host "SAS URL generated."
 
+# 8) update artifacts.json in-place
+$artifactsPath = Join-Path $repoRoot "artifacts.json"
+$artifactsObj = [ordered]@{ resourcesTemplate = $sasUrl }
+($artifactsObj | ConvertTo-Json) | Set-Content -Path $artifactsPath -Encoding UTF8
+Write-Host "Updated artifacts.json"
 
-# updating artifacts config
-Write-Output "Updating artifacts config"
-$artifactsConfig = @{
-    resourcesTemplate = "$resourcesTemplateURL"
+# 9) robust git commit
+Push-Location $repoRoot
+try {
+  if (-not (Get-Command git -ErrorAction SilentlyContinue)) {
```

The script attempts a git commit but will silently skip committing if `git` is missing or the repo check fails. The task requirement states that `artifacts.json` changes must be committed as part of the submission. Consider one of these approaches: 1) If commit cannot be performed, exit with a non-zero error and a clear message so the user must commit before submitting; or 2) If you prefer not to fail, at minimum print an unmistakable, prominent instruction explaining that `artifacts.json` was updated but not committed and how to commit it. This reduces chance of someone submitting without the required committed change.